Gateway/Request/CheckoutDataBuilder.php

```diff
@@ -95,9 +106,19 @@ public function build(array $buildSubject)
         // do not send email
         $order->setCanSendNewEmailFlag(false);
 
-        $requestBody['returnUrl'] = $this->storeManager->getStore()->getBaseUrl(
-                \Magento\Framework\UrlInterface::URL_TYPE_LINK
-            ) . 'adyen/transparent/redirect?merchantReference=' . $order->getIncrementId();
+        $pwaOrigin = $this->adyenHelper->getAdyenAbstractConfigData(
+            "payment_origin_url",
+            $this->storeManager->getStore()->getId()
+        );
+
+        if ($pwaOrigin) {
+            $returnUrl = $pwaOrigin . 'adyen/process/result?merchantReference=' . $order->getIncrementId();
```

Comment:
Maybe some merchants add or forget adding the trailing slash in the config value
```suggestion
            $returnUrl = rtrim($pwaOrigin, '/') . '/adyen/process/result?merchantReference=' . $order->getIncrementId();
```